scheduler/reconciler/reconcile_node.go

```diff
@@ -157,8 +157,12 @@ func (nr *NodeReconciler) computeForNode(
 		deploymentFailed = nr.DeploymentCurrent.Status == structs.DeploymentStatusFailed
 	}
 
-	// Track desired total placements across all loops
-	var desiredTotal int
+	// Track desired total and desired canaries across all loops
+	desiredTotal := map[string]int{}
+	desiredCanaries := map[string]int{}
+
+	// Track whether we're during a canary update
+	var isCanarying bool
```

Comment:
Don't we need to track `isCanarying` separately for each task group? One might have canaries and the other doesn't.